File: CategoryA/1542A.c
#include <stdio.h>

int main() {
    int t;
    scanf("%d", &t);

    while (t--) {
        int n;
        scanf("%d", &n);

        int pares = 0, impares = 0;
        int x;

        for (int i = 0; i < 2 * n; i++) {
            scanf("%d", &x);
            if (x % 2 == 0) {pares++;
            } 
			else {impares++;
            }
        }
        if (pares == n && impares == n) {printf("Yes\n");
        } 
		else {printf("No\n");
        }
    }
    return 0;
}

File: CategoryA/1542A.py
t = int(input())

for _ in range(t):
    n = int(input())
    lista = list(map(int, input().split()))

    pares = 0
    impares = 0

    for x in lista:
        if x % 2 == 0:
            pares += 1
        else:
            impares += 1

    if pares == n and impares == n:
        print("Yes")
    else:
        print("No")

